react-components/src/components/SignIn/useSignin.js

```diff
@@ -66,6 +70,8 @@ export const useSignin = props => {
             // 3. merge the shopping cart if necessary
             let mergedCartId;
 
+            // Get cartId from localStorage, not from cookie
+            const cartId = storage.getItem('cartId');
```

Comment:
You could also use a helper from peregrine for that https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/store/actions/cart/asyncActions.js#L403-L413

Reply:
👍